### PR TITLE
Adding FragmentKey type.

### PR DESCRIPTION
**Pull request by Laimiux**
## What
Adding `FragmentKey` type which only contains `tag` and no UI rendering logic which means it has no generic parameter.
```kotlin
interface FragmentKey : Parcelable {
    abstract val tag: String
}
```

It's a pre-cursor to https://github.com/instacart/formula/pull/181

### CODE REVIEW

**Comment by carrotkite** [comment]
### JaCoCO Code Coverage 76.44% :white_check_mark:
| Class | Covered | Meta | Status |
|:---|:---:|:---:|:---:|
| `com/instacart/formula/fragment/FragmentFlowState` | 0% | 0% | :white_check_mark: |
| `com/instacart/formula/fragment/FragmentFlowStore` | 100% | 0% | :white_check_mark: |
| `com/instacart/formula/fragment/FragmentLifecycle` | 70% | 0% | :white_check_mark: |
| `com/instacart/formula/fragment/FragmentEnvironment` | 50% | 0% | :white_check_mark: |
| `com/instacart/formula/fragment/FragmentContract` | 100% | 0% | :white_check_mark: |
| `com/instacart/formula/fragment/FormulaFragment` | 50% | 0% | :white_check_mark: |
| `com/instacart/formula/integration/internal/SingleBinding` | 100% | 0% | :white_check_mark: |
| `com/instacart/formula/integration/internal/BaseBindingBuilder` | 100% | 0% | :white_check_mark: |
| `com/instacart/formula/integration/internal/CompositeBinding` | 100% | 0% | :white_check_mark: |
| `com/instacart/formula/android/internal/ActivityStoreContextImpl` | 80% | 0% | :white_check_mark: |
| `com/instacart/formula/FormulaAndroid` | 50% | 0% | :white_check_mark: |
| `com/instacart/formula/integration/FlowIntegration` | 100% | 0% | :white_check_mark: |
| `com/instacart/formula/integration/FragmentFlowRenderView` | 64% | 0% | :white_check_mark: |
| `com/instacart/formula/integration/Binding` | 100% | 0% | :white_check_mark: |
| `com/instacart/formula/integration/FragmentBindingBuilder` | 100% | 0% | :white_check_mark: |
| `com/instacart/formula/integration/FlowDeclaration` | 25% | 0% | :white_check_mark: |
| `com/instacart/formula/integration/KeyState` | 100% | 0% | :white_check_mark: |
| `com/instacart/formula/integration/Bindings` | 100% | 0% | :white_check_mark: |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: formula-android/src/main/java/com/instacart/formula/fragment/FormulaFragment.kt [hunk 1]
```diff
@@ -113,7 +113,7 @@ class FormulaFragment<RenderModel> : Fragment(), BaseFormulaFragment<RenderModel
         return stateRelay.value
     }
 
-    override fun getFragmentContract(): FragmentContract<RenderModel> {
+    override fun getFragmentKey(): FragmentContract<RenderModel> {
```
Comment (Jawnnypoo) [inline formula-android/src/main/java/com/instacart/formula/fragment/FormulaFragment.kt:116]: Is this the right return type?

**Comment by Laimiux** [comment]
> LGTM, maybe in the release notes, a migration guide?

Currently, it's more of an internal type so there isn't anything we need to migrate.

File: formula-android/src/main/java/com/instacart/formula/fragment/FormulaFragment.kt (hunk 1 above)
Comment (Laimiux) [reply formula-android/src/main/java/com/instacart/formula/fragment/FormulaFragment.kt]: Good call, will update to `FragmentKey`.

**Pull request merged**